libstorage/src/Volume.cc

```diff
@@ -2186,17 +2199,18 @@ EncryptType Volume::detectEncryption()
 			break;
 		    }
 		bool excTime, excLines;
-		c.executeRestricted( cmd, 15, 500, excTime, excLines );
-		bool ok = c.retcode()==0 || (excTime && !excLines);
-		y2mil("ok:" << ok << " retcode:" << c.retcode() << " excTime:" << excTime <<
-		      " excLines:" << excLines);
+                SystemCmd fsckCmd;
+                fsckCmd.executeRestricted( cmd, 15, 500, excTime, excLines );
+                bool ok = fsckCmd.retcode()==0 || (excTime && !excLines);
```

Comment:
Well, yes, there are many things that could be improved in that old code, but the approach here is to keep the diff minimalistic. This considerably improves the chances that the patch cleanly applies with increasingly older releases.